### PR TITLE
Fix type assertion error in init and join commands for kubeadm

### PR DESCRIPTION
**Pull request by yxxhero**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Fix type assertion error in init and join commands for kubeadm  

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by yxxhero** [comment]
/release-note-edit

```
kubeadm: fix type assertion error in init and join commands.
```

**Review by neolit123 — COMMENTED** [review]
/release-note-none

init and join are not the only kubeadm commands where workflow.NewRunner is used.

**Comment by neolit123** [comment]
/release-note-edit

```release-note
NONE
```

**Comment by yxxhero** [comment]
@neolit123 Thanks for your review.

**Comment by yxxhero** [comment]
@neolit123 any updates? thanks so much.

**Comment by neolit123** [comment]
> @neolit123 any updates? thanks so much.

this comment is not addressed:
>> init and join are not the only kubeadm commands where workflow.NewRunner is used.

**Comment by yxxhero** [comment]
> > @neolit123 any updates? thanks so much.
> 
> this comment is not addressed:
> 
> > > init and join are not the only kubeadm commands where workflow.NewRunner is used.

The other places have been safely handled. thanks so much. please ping me if I missed something. @neolit123 thanks so much again.

File: cmd/kubeadm/app/cmd/init.go [hunk 1]
```diff
@@ -118,7 +118,11 @@ func newCmdInit(out io.Writer, initOptions *initOptions) *cobra.Command {
 				return err
 			}
 
-			data := c.(*initData)
+			data, ok := c.(*initData)
+			if !ok {
+				return errors.New("init command invoked with an invalid data struct")
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/init.go:123]:
```suggestion
				return errors.New("invalid data struct")
```

File: cmd/kubeadm/app/cmd/join.go [hunk 2]
```diff
@@ -175,7 +175,10 @@ func newCmdJoin(out io.Writer, joinOptions *joinOptions) *cobra.Command {
 				return err
 			}
 
-			data := c.(*joinData)
+			data, ok := c.(*joinData)
+			if !ok {
+				return errors.New("join command invoked with an invalid data struct")
```
Comment (neolit123) [inline cmd/kubeadm/app/cmd/join.go:180]:
```suggestion
				return errors.New("invalid data struct")

**Review by neolit123 — COMMENTED** [review]
let's just use this same text that was already in cmd/reset.go
please keep the commits squashed to 1.

**Comment by yxxhero** [comment]
@neolit123 thanks so much.

**Review by neolit123 — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**